Alerts round 2

senior opened the pull request:
Fixes up notification issues with the alert lifecycle and adds special handling for alerts to the activity feed. The descriptions on the individual commits have details on the notification changes.

iethree committed:
Add alerts to the activity feed

Previously alerts were showing up as pulses. This commit adds a
distinction between pulses and alerts

iethree committed:
When an admin deletes their own alert, don't send an email

Previously whenever an alert was deleted it would notify the creator
of the deletion, even if the creator was the one that did the
deleting. This commit changes that behavior to only notify if the
creator is different from the user deleting the alert.

iethree committed:
Recipients added at alert creation should be notified

Previously the creator of the alert would be notified, but any
recipients added to the new alert were not being notified. This commit
continues to notify the creator, but then will let all of the other
recipients know that they've been subscribed to an alert.

iethree committed:
Fix for a pulse_channel_test failing due to resultset order

The correct data was coming back, just in a different order on some
databases. Put the rows into a set so that it's not sensitive to the
arbitrary order.

iethree committed:
Grab the visualization settings from the first result for the goal value

salsakran reviewed (approved):
frontend code looks reasonable enough

mazameli committed:
get rid of email card border

iethree committed:
Refactor pulse and alert tests

Specifically this asserts more on the contents of the email (such as
the goal that was reached) and uses the email regexing code in
metabase.email-test which has been expanded to work on multipart
emails (like pulses and alerts).
